src/java/org/apache/nutch/tools/FreeGenerator.java

```diff
@@ -66,79 +64,80 @@
   private static final String FILTER_KEY = "free.generator.filter";
   private static final String NORMALIZE_KEY = "free.generator.normalize";
 
-  public static class FG extends MapReduceBase implements
-      Mapper<WritableComparable<?>, Text, Text, Generator.SelectorEntry>,
-      Reducer<Text, Generator.SelectorEntry, Text, CrawlDatum> {
-    private URLNormalizers normalizers = null;
-    private URLFilters filters = null;
-    private ScoringFilters scfilters;
-    private CrawlDatum datum = new CrawlDatum();
-    private Text url = new Text();
-    private int defaultInterval = 0;
+  public static class FG {
+    private static URLNormalizers normalizers = null;
+    private static URLFilters filters = null;
+    private static ScoringFilters scfilters;
+    private static CrawlDatum datum = new CrawlDatum();
+    private static Text url = new Text();
+    private static int defaultInterval = 0;
```

Comment:
See comments in Generator etc. regarding static variables shared between mapper and reducer classes.